uaclient/cli.py

```diff
@@ -248,21 +251,51 @@ def status_parser(parser):
     return parser
 
 
-@assert_root
-@require_valid_entitlement_name("disable")
-@assert_attached(ua_status.MESSAGE_ENABLE_FAILURE_UNATTACHED_TMPL)
-def action_disable(args, cfg):
+def _perform_disable(entitlement_name, cfg, *, assume_yes):
     """Perform the disable action on a named entitlement.
 
-    @return: 0 on success, 1 otherwise
+    :param entitlement_name: the name of the entitlement to enable
+    :param cfg: the UAConfig to pass to the entitlement
+    :param assume_yes:
+        Assume a yes response for any prompts during service enable
+
+    @return: True on success, False otherwise
     """
-    ent_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME[args.name]
-    entitlement = ent_cls(cfg, assume_yes=args.assume_yes)
-    ret = 0 if entitlement.disable() else 1
+    ent_cls = entitlements.ENTITLEMENT_CLASS_BY_NAME[entitlement_name]
+    entitlement = ent_cls(cfg, assume_yes=assume_yes)
+    ret = entitlement.disable()
     cfg.status()  # Update the status cache
     return ret
 
 
+@assert_root
+@require_valid_entitlement_names("disable")
```

Comment:
We can drop this decorate and call get_valid_entitlement_names (which should probably accept an allow_beta param.